activemodel/test/cases/attribute_assignment_test.rb

```diff
@@ -95,9 +95,11 @@ def dup
   end
 
   test "an ArgumentError is raised if a non-hash-like object is passed" do
-    assert_raises(ArgumentError) do
+    err = assert_raises(ArgumentError) do
       Model.new(1)
     end
+
+    assert_equal("When assigning attributes, you must pass a hash as an argument, #{1.class} passed.", err.message)
```

Comment:
`1.class` was only necessary for compatibility with Ruby 2.3; master now requires Ruby 2.4 or newer, so you can just use `Integer` here.